net/libfab/libfab.c

```diff
@@ -0,0 +1,428 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+/**
+ * Mapping for m0net-->libfabric API
+ * xo_dom_init                    =  fi_getinfo, fi_fabric, fi_domain
+ * xo_dom_fini                    =  fi_close, // free resources
+ * xo_tm_init                     =  // Empty function ? fi_send
+ * xo_tm_confine                  =
+ * xo_tm_start                    =  // fi_send
+ * xo_tm_stop                     =  fi_cancel
+ * xo_tm_fini                     =  // Empty function
+ * xo_end_point_create            =  fi_endpoint, fi_pep, fi_av, fi_cq, fi_cntr, fi_eq, fi_bind(av/cq/cntr/eq), fi_pep_bind
+ * xo_buf_register                =  fi_mr_reg, fi_mr_desc, fi_mr_key, fi_mr_bind, fi_mr_enable
+ * xo_buf_deregister              =  fi_close
+ * xo_buf_add                     =  fi_send/fi_recv
+ * xo_buf_del                     =  fi_cancel 
+ * xo_bev_deliver_sync            =  Is it needed?
+ * xo_bev_deliver_all             =  Is it needed?
+ * xo_bev_pending                 =  Is it needed?
+ * xo_bev_notify                  =  Is it needed?
+ * xo_get_max_buffer_size         =  // need to define new API
+ * xo_get_max_buffer_segment_size =  // need to define new functions
+ * xo_get_max_buffer_segments     =  // need to define new functions
+ * xo_get_max_buffer_desc_size    =  
+ * 
+ */
+
+/**
+ * @addtogroup netlibfab
+ *
+ * @{
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_NET
+#include "lib/trace.h"
+
+#include "net/net.h"
+#include "lib/bitmap.h"
+#include "lib/chan.h"
+#include "lib/memory.h"
+#include "libfab.h"
+#include <errno.h>
+
+
+/** Network transfer machine */
+struct transfer_ma {
+	/**
+	* Generic transfer machine with buffer queues, etc. 
+	*/
+	struct m0_net_transfer_mc *t_ma;
+
+	/**
+	* TODO: Is poller thread required ?
+	*/
+};
+
+/** Used as m0_net_xprt_ops::xo_dom_init(). */
+static int libfab_dom_init(struct m0_net_xprt *xprt, struct m0_net_domain *dom)
+{
+	struct libfabric_domain_params *fab;
+	int rc;
+
+	M0_ENTRY();
+	
+	M0_ALLOC_PTR(fab);
+	if (fab == NULL)
+		return M0_RC(-ENOMEM);
+
+	fab->hints = fi_allocinfo();
+	if (fab->hints != NULL) {
+		// fab->hints->ep_attr->type = FI_EP_RDM;
+		// fab->hints->caps = FI_MSG;
+		// fab->hints->fabric_attr->prov_name = "verbs";
+		rc = fi_getinfo(FI_VERSION(FI_MAJOR_VERSION,FI_MINOR_VERSION),
+				NULL, NULL, 0, fab->hints, &fab->fi);
+		if (rc == FI_SUCCESS ) {
+			rc = fi_fabric(fab->fi->fabric_attr, &fab->fabric,
+				       NULL);
+			if (rc == FI_SUCCESS ) {
+				rc = fi_domain(fab->fabric, fab->fi,
+					       &fab->domain, NULL);
+				if (rc == FI_SUCCESS )
```

Comment:
Please follow coding-style: no c++ comments, spacing, etc. This will save us a lot of time.